drv/lpc55-update-server/src/main.rs

```diff
@@ -963,6 +1343,18 @@ fn copy_from_flash_range(
     Ok(())
 }
 
+fn bootstate() -> Result<RotBootStateV2, HandoffDataLoadError> {
+    // Safety: Data is published by stage0
+    let addr = unsafe { BOOTSTATE.assume_init_ref() };
+    RotBootStateV2::load_from_addr(addr)
+}
+
+fn round_up_to_flash_page(offset: u32) -> Option<u32> {
+    offset
+        .checked_add(BYTES_PER_FLASH_PAGE as u32 - 1)
+        .map(|sum| sum & !(BYTES_PER_FLASH_PAGE as u32 - 1))
+}
```

Comment:
Can we use https://doc.rust-lang.org/stable/std/primitive.u32.html#method.checked_next_multiple_of ? This is a new stable API